tests/python/pants_test/backend/jvm/tasks/jvm_compile/zinc/zinc_compile_integration_base.py

```diff
@@ -180,3 +193,128 @@ def test_pool_created_for_fresh_compile_but_not_for_valid_compile(self):
                             'testprojects/src/scala/org/pantsbuild/testproject/javasources')
 
       self.assertNotIn('isolation-zinc-pool-bootstrap', second_run.stdout_data)
+
+  def test_source_compat_binary_incompat_scala_change(self):
+    with temporary_dir() as cache_dir, \
+      self.temporary_workdir() as workdir, \
+      temporary_dir(root_dir=get_buildroot()) as src_dir:
+
+      config = {
+        'cache.compile.zinc': {'write_to': [cache_dir], 'read_from': [cache_dir]},
+        'compile.zinc': {'incremental_caching': True },
```

Comment:
Hm... incremental_caching shouldn't be necessary to catch this, right? Should be possible to repro without the cache at all (at least according to the report).